### PR TITLE
#6588: Refactor moreh_matmul interface

### PR DESCRIPTION
**Pull request by seunghwan100**
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8495434408

### CODE REVIEW

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py [hunk 1]
```diff
@@ -229,6 +236,8 @@ def test_moreh_matmul_backward(params, input_b1, input_b2, other_b1, other_b2, r
         logger.debug(f"other_grad passing={passing}")
         logger.debug(f"other_grad pcc={output_pcc}")
         assert passing
+    else:
+        assert tt_other_grad is None
 
 
 @skip_for_wormhole_b0()
```
Comment (dongjin-na) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py:243]: I would appreciate it if you could remove `@skip_for_wormhole_b0()` from all tests.

Comment (seunghwan100) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py]: > I would appreciate it if you could remove `@skip_for_wormhole_b0()` from all tests.

I removed all ```@skip_for_wormhole_b0()```

Comment (eyonland) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py]: Thanks for this example related to #5044

**Review by dongjin-na — APPROVED** [review]
LGTM.

**Review by eyonland — APPROVED** [review]

**Comment by dongjin-na** [comment]
commits merged into the main branch by [#8388](https://github.com/tenstorrent/tt-metal/pull/8388)

**Pull request closed without merging**